packages/pigeon/platform_tests/ios_swift_unit_tests/ios/RunnerTests/Utils.swift

```diff
@@ -0,0 +1,59 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Foundation
+
+func equals(_ x: Any?, _ y: Any?) -> Bool {
+  if x == nil, y == nil {
+    return true
+  }
+  
+  guard let x = x as? AnyHashable,
+        let y = y as? AnyHashable else {
+          return false
+        }
+  return x == y
+}
+
+func equalsList(_ x: [Any?]?, _ y: [Any?]?) -> Bool {
+  if x == nil, y == nil {
+    return true
+  }
+  
+  guard x?.count == y?.count else { return false }
+  guard let x = x, let y = y else {
+    return false
+  }
+  
+  for i in 0..<(x.count) {
+    if equals(x[i], y[i]) == false {
+      return false
+    }
+  }
+  
+  return true
+}
+
+func equalsDictionary(_ x: [AnyHashable: Any?]?, _ y: [AnyHashable: Any?]?) -> Bool {
+  if x == nil, y == nil {
+    return true
+  }
+  
+  guard x?.count == y?.count else { return false }
+  guard let x = x, let y = y else {
+    return false
+  }
+  
+  for (key, valueX) in x {
+    if let valueY = y[key] {
+      if equals(valueX, valueY) == false {
+        return false
+      }
+    } else {
+      return false
+    }
+  }
```

Comment:
`allSatisfy`

Reply:
Would it look like this?
```swift
return x.allSatisfy { equals($0.value, y[$0.key]) }
```
This emit a warning with the message:
**Expression implicitly coerced from 'Any??' to 'Any?'**

Reply:
This means it's wrapped by double optional. You can use `as Any?`.  

That being said, this shouldn't happen in the first place. There is no benefit of having a dictionary of type `[AnyHashable: Any?]` instead of just `[AnyHashable: Any]`

Reply:
It's needed because we have this situations:
```dart
List<List<bool?>?>? nestedList;
Map<String?, String?>? mapWithAnnotations;
Map<String?, Object?>? mapWithObject;
```

Reply:
> can you elaborate more where do they come from? Why do we use optional as the map value?

Pigeon APIs are defined by the clients of Pigeon; there's no "we" in this context. (Currently Pigeon only supports optional generic types, for somewhat complicated reasons, but even once that's eventually addressed people could still define their APIs this way if they wanted to.)

Reply:
Done